eventbridge-events-replay-cdk/python/stack.py

```diff
@@ -0,0 +1,340 @@
+from aws_cdk import (
+    Stack,
+    aws_kinesis as kinesis,
+    Duration,
+    aws_stepfunctions as stepfunctions,
+    aws_stepfunctions_tasks as tasks,
+    aws_lambda,
+    aws_events as events,
+    aws_iam,
+    Aws,
+    aws_pipes,
+    CfnOutput,
+)
+import os
+from constructs import Construct
+
+dirname = os.path.dirname(__file__)
+
+
+class EventbridgeEventsReplayStack(Stack):
+    def __init__(
+        self,
+        scope: Construct,
+        construct_id: str,
+        cross_account_eventbus_arn: str,
+        **kwargs,
+    ) -> None:
+        super().__init__(scope, construct_id, **kwargs)
+
+        # kinesis stream to hold  replayed and sampled events
+        replayed_events_stream = kinesis.Stream(
+            self, "ReplayedEventsStream", stream_name="ReplayedEventsStream"
+        )
+
+        # Main event bus with actual business transactions
+        main_bus = events.EventBus(
+            self, "MainEventBus", event_bus_name="EventBusWithBizTransactions"
+        )
+
+        main_bus_archive = main_bus.archive(
+            "main_bus_archive",
+            archive_name="ArhiveOfMainBus",
+            description="EventBusWithBizTransactions Archive",
+            event_pattern=events.EventPattern(region=[Aws.REGION]),
+            retention=Duration.days(30),
+        )
+
+        event_pattern = {"replay-name": [{"exists": True}]}
+
+        kinesis_write_role = aws_iam.Role(
+            self,
+            "KinesisWriteRole",
+            assumed_by=aws_iam.ServicePrincipal(f"events.amazonaws.com"),
+            role_name="kinesis_write_role_for_events_rule",
+        )
+        kinesis_write_role.attach_inline_policy(
+            aws_iam.Policy(
+                self,
+                "write_to_kinesis_policy",
+                statements=[
+                    aws_iam.PolicyStatement(actions=["kinesis:*"], resources=["*"])
```

Comment:
can you make this permission only to the kinesis stream that you are going to use?